### DISCUSSION TITLE
Getting error 259 on random files

### DISCUSSION THREAD

**Discussion by chuddale** [s0hv/foo_discord_rich#9] [Q&A] [answered]
_"Discord Rich Presence Integration v1.3.0-beta-3+c4b3f11: artwork uploader exited with status: 259 and error:_ 

Error received multiple times throughout different files. It seems like a diceroll on whether it happens or not. For some reason it seems more likely to happen on files within newer directories. Has happened on files with 2mb cover metadata, so size doesn't seem to be the issue. 
Manually entering a valid imgur url to the photo leaves the cover as a transparent square on discord status. Re-generating a URL returns the same 259 error.

Tested both on imgur-uploader.exe as well as foobar2000-catbox.exe

**Comment by s0hv** [comment]
After further inspection it seems that the 259 error code is a special error code that means that the process did not exit and is still alive https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getexitcodeprocess I'll create some mitigations for this in the next release

**Comment by s0hv** [comment, marked as answer]
Try out the [beta 4 release](https://github.com/s0hv/foo_discord_rich/releases/tag/1.3.0-beta-4). It should now give a more descriptive error message on error 259 "Process timed out". If this seems to happen often go to the plugin preferences and modify the process timeout found on the advanced tab. Setting it to 0 sets the timeout to 1 day (prefer something lower like 60 seconds. That way the process is not left open for too long in case something causes it to get stuck), otherwise the value provided is used which is 10 seconds by default. 

The root cause of this is probably low upload speeds combined with large files which causes the uploading to longer than the timeout.

**Reply by chuddale** [reply]
That seems to be the fix. Thank you so much! And thank you for the quick response.
